natlas-agent/natlas/screenshots.py

```diff
@@ -43,64 +102,41 @@ def get_web_screenshots(target, scan_id, proctimeout):
         logger.warning(f"TIMEOUT: Killing aquatone against {target}")
         process.kill()
 
-    session_path = os.path.join(outFiles, "aquatone_session.json")
-    if not os.path.isfile(session_path):
-        return output
-
-    with open(session_path) as f:
-        session = json.load(f)
-
-    if session["stats"]["screenshotSuccessful"] > 0:
-        logger.info(
-            f"{target} - Success: {session['stats']['screenshotSuccessful']}, Fail: {session['stats']['screenshotFailed']}"
-        )
-
-        for k, page in session["pages"].items():
-            fqScreenshotPath = os.path.join(outFiles, page["screenshotPath"])
-            if page["hasScreenshot"] and os.path.isfile(fqScreenshotPath):
-                urlp = urlparse(page["url"])
-                if not urlp.port and urlp.scheme == "http":
-                    port = 80
-                elif not urlp.port and urlp.scheme == "https":
-                    port = 443
-                else:
-                    port = urlp.port
-                logger.info(
-                    f"{urlp.scheme.upper()} screenshot acquired for {page['hostname']} on port {port}"
-                )
-                output.append(
-                    {
-                        "host": page["hostname"],
-                        "port": port,
-                        "service": urlp.scheme.upper(),
-                        "data": base64_image(fqScreenshotPath),
-                    }
-                )
-    return output
+    return parse_aquatone_session(output_dir)
 
 
 def get_vnc_screenshots(target, scan_id, proctimeout):
 
     scan_dir = utils.get_scan_dir(scan_id)
-    outFile = os.path.join(scan_dir, f"vncsnapshot.{scan_id}.jpg")
+    output_file = os.path.join(scan_dir, f"vncsnapshot.{scan_id}.jpg")
```

Comment:
Oh yeah this would be much nicer if we passed in a scan object and it had a method like `scan.get_output_file("screenshot.jpg")`

An idea for your agent refactor.

Reply:
Yeah the refactor breaks output up into per-plugin folders and there is a shared utility module that enables a plugin to get it's output folder by scan_id and plugin_name, which consolidates basically all of this output logic into a common set of functions. From there it'll be easy to add in a function for the scan object